cli/cmd/plugin/ui/web/tanzu-ui/src/views/unmanaged-cluster/UnmanagedClusterInventory.tsx

```diff
@@ -5,35 +5,48 @@ import { useNavigate } from 'react-router-dom';
 // Library imports
 import { CdsButton } from '@cds/react/button';
 import { CdsIcon } from '@cds/react/icon';
-import { CdsAlert, CdsAlertGroup } from '@cds/react/alert';
 import { CdsModal, CdsModalHeader, CdsModalActions, CdsModalContent } from '@cds/react/modal';
 
 // App imports
 import { NavRoutes } from '../../shared/constants/NavRoutes.constants';
+import PageNotification, { Notification, NotificationStatus } from '../../shared/components/PageNotification/PageNotification';
 import { UnmanagedCluster } from '../../swagger-api';
 import { UnmanagedService } from '../../swagger-api/services/UnmanagedService';
 import UnmanagedClusterCard from './UnmanagedClusterCard/UnmanagedClusterCard';
 import './UnmanagedClusterInventory.scss';
 
 function UnmanagedClusterInventory() {
+    const [showLoading, setShowLoading] = useState<boolean>(false);
+    const [notification, setNotification] = useState<Notification | null>(null);
     const [unmanagedClusters, setUnmanagedClusters] = useState<UnmanagedCluster[]>([]);
     const [showDeleteModal, setShowDeleteModal] = useState<boolean>(false);
     const [clusterNameForDeletion, setClusterNameForDeletion] = useState<string>('');
     const navigate = useNavigate();
 
-    function retrieveUnmanagedClusters() {
-        UnmanagedService.getUnmanagedClusters().then(
-            (data) => {
-                setUnmanagedClusters(data);
-            },
-            (err) => {
-                console.warn(`unmanaged clusters get api failed with error: ${err}`);
-            }
-        );
+    // sets notification to null to dismiss alert
+    function dismissAlert() {
+        setNotification(null);
     }
 
+    const retrieveUnmanagedClusters = async () => {
+        setShowLoading(true);
+        try {
+            const data = await UnmanagedService.getUnmanagedClusters();
+            setUnmanagedClusters(data);
+        } catch (e) {
+            setNotification({
+                status: NotificationStatus.DANGER,
+                message: `Unable to retrieve Unmanaged Clusters: ${e}`,
+            } as Notification);
+        } finally {
+            setShowLoading(false);
+        }
+    };
+
     useEffect(() => {
-        retrieveUnmanagedClusters();
+        retrieveUnmanagedClusters().then(() => {
```

Comment:
same here (please use async await. you may need to create the function inside the scope and call it.)